add api key to sidebar in stacks-blockchain-api,ordinals-api,token-me… (#381)
hirosystems/docs: docs/api/how-to-use-api-key.md

---
Title: How to use API key
---

> # How to use API Key?

Reviewer: ```suggestion
# How to use API Key
```

So that the title is consistent with other how-to guides. Also, the question mark won't show up on the left-nav.

API keys are alpha-numeric codes that identify and authenticate an application or developer. You can use API keys to control access to your API calls.

# Steps to use API Key

This guide helps you with the steps to use the API key to interact with the API.

> > **_NOTE:_**

Reviewer: Perhaps this can be a warning instead of note?

> > **_Warning:_**

Reviewer: ```suggestion
> :::caution
```

Suggesting the use of caution admonition: https://docusaurus.io/docs/next/markdown-features/admonitions

> > The API key is passed in the **header** of your API call and is used only for private use, like server-side applications. This key is not to be exposed or shared.

Reviewer: ```suggestion
> The API key is passed in the **header** of your HTTP API request and is used only for private use, like in server-side applications. This key is not to be exposed or shared; if you use the API key in your client-side application (e.g., frontend and browser-based applications), attackers can capture it using the client tools (e.g., browser console) and abuse your API key.
```

There are multiple ways to interact with the API endpoints. In this guide, we will walk through the following approaches with the help of an API endpoint.

- cURL

> - Typescript function

Reviewer: ```suggestion
- Typescript client
```

## Using cURL

Using cURL, you will pass the API key in an `x-hiro-api-key` header. Use the following command as an example to call the API endpoint `https://api.hiro.so/<your-api-endpoint>`.

`curl https://api.hiro.so/... -H 'x-hiro-api-key: <your-API-value>'`

## Using the API key in Typescript

> You can use the following sample to pass the API key in your function using Typescript.

Reviewer: ```suggestion
The following Typescript snippet demonstrates the instantiation of  RESTful HTTP requests with API key.
```

---
Title: How to use API key
---
# How to use API Key

API keys are alpha-numeric codes that identify and authenticate an application or developer.

# Steps to use API Key

This guide helps you with the steps to use the API key to interact with the API.

> **_NOTE:_**
> The API key is passed in the **header** of your API call and is used only for private use, like server-side applications. This key is not to be exposed or shared.

> There are multiple ways to interact with the API endpoints. However, this document walks you through three ways to make an [API call](https://docs.hiro.so/api#tag/Microblocks/operation/get_microblock_list). You can choose between any of the following methods.

Reviewer: In the preview, this is how it looks:
<img width="858" alt="image" src="https://github.com/hirosystems/docs/assets/1562654/cc3cf2f7-32f8-472d-92e7-7baa8fa57bee">

And it can a bit more clearer.

Maybe:

> There are multiple ways to interact with the API endpoints. In this guide, we will walk through the following approaches with the help of an API endpoint.

- Curl
- Postman
- Typescript function

## Using Curl

> Using Curl, you will pass the API key in an `x-hiro-api-key` header. Use the following command as an example to call the API endpoint `https://api.hiro.so/extended/v1/microblock`.

Reviewer: Since we are including the same guide across 3 different API products, the API endpoint example in use could be confusing in Metadata and Ordinals API guides.

Any way we can parameterize the URL, so a different value can be plugged at Docs' build time for different API guides?

If the above is not possible, maybe we could generalize the URL; something like.. `https://api.hiro.so/ ... ` 

Basically, not a fully qualified Stacks Blockchain API as an example, but just a pattern. If you choose this method, then you may have to update the screenshots, also.
